cloudify_aws/ec2/resources/transit_gateway.py

```diff
@@ -0,0 +1,308 @@
+        elif iface.status in UNAVAILABLE + FAILED:
+            raise NonRecoverableError(
+                'The {r} creation request '
+                'results in a fatal error: {s}'.format(
+                    r=TG_ATTACHMENT,
+                    s=iface.status))
+        else:
+            request = {
+               TG_ATTACHMENT_ID: transit_gateway_attachment_id
+            }
+            try:
+                iface.accept(request)
+            except (NonRecoverableError, ClientError) as e:
+                raise OperationRetry(
+                    'Waiting for {t} to be in valid state: {s}. '
+                    'Error={e}'.format(t=transit_gateway_attachment_id,
+                                       s=iface.status,
+                                       e=e))
+
+    request = {
+        TG_ID: transit_gateway_id,
+        'VpcId': vpc_id,
+        'SubnetIds': subnet_ids
+    }
+
+    response = iface.create(request)
+    ctx.logger.info('Sent the {r} creation request.'.format(
+        r=TG_ATTACHMENT))
+    ctx.source.instance.runtime_properties[TG_ATTACHMENTS][vpc_id] = \
+        utils.JsonCleanuper(response).to_dict()
+
+
+@decorators.aws_relationship(EC2TransitGatewayAttachment, RESOURCE_TYPE)
+def delete_vpc_attachment(ctx, iface, transit_gateway_attachment_id=None, **_):
+    transit_gateway_attachment_id = transit_gateway_attachment_id or \
+        get_attachment_id_from_runtime_props(ctx)
+    if not transit_gateway_attachment_id:
+        ctx.logger.error('No transit_gateway_attachment_id was provided. '
+                         'Skipping delete attachment.')
+        return
+    iface = EC2TransitGatewayAttachment(
+        ctx.source.node,
+        transit_gateway_attachment_id,
+        iface.client,
+        ctx.logger)
+    request = {
+        TG_ATTACHMENT_ID: transit_gateway_attachment_id
+    }
+    if iface.status == 'deleting':
+        raise OperationRetry(
+            'The {r} deletion request has been received and is processing. '
+            'State: {s}.'.format(r=TG_ATTACHMENT, s=iface.status))
+    elif iface.status in UNAVAILABLE:
+        ctx.logger.info('The {r} has been deleted.'.format(
+            r=TG_ATTACHMENT))
+        return
+    iface.delete(request)
+    raise OperationRetry(
+        'Sent the {r} deletion request.'.format(r=TG_ATTACHMENT))
+
+
+def get_attachment_id(props):
+    attachment = props.get(TG_ATTACHMENT, {})
+    return attachment.get(TG_ATTACHMENT_ID)
+
+
+def get_attachment_id_from_runtime_props(ctx):
+    vpc_id = ctx.target.instance.runtime_properties.get(
+        constants.EXTERNAL_RESOURCE_ID)
+    if TG_ATTACHMENTS in ctx.source.instance.runtime_properties:
+        if vpc_id in ctx.source.instance.runtime_properties[TG_ATTACHMENTS]:
+            return ctx.source.instance.runtime_properties[
+                TG_ATTACHMENTS][vpc_id][TG_ATTACHMENT][TG_ATTACHMENT_ID]
+    else:
+        ctx.source.instance.runtime_properties[TG_ATTACHMENTS] = {}
```

Comment:
so if you don't find the vpc you are setting all attachments to {}?
if there is an attachment for another vpc(if it can happen) so you will delete it.

Reply:
This is for use by all attachments (we can have many.)